src/msw/printdlg.cpp

```diff
@@ -423,6 +423,12 @@ void wxWindowsPrintNativeData::InitializeDevMode(const wxString& printerName, Wi
     // this replaces the PrintDlg function which creates the DEVMODE filled only with data from default printer.
     if ( !m_devMode && !printerName.IsEmpty() )
     {
+        // ensure that we have a printer object here, otherwise we are unable to determine m_devMode
+        WinPrinter fallbackPrinter;
+        if (!printer) {
+            printer = &fallbackPrinter;
+        }
+
         // Open printer
         if ( printer && printer->Open( printerName ) == TRUE )
```

Comment:
This check doesn't make sense any more now:
```suggestion
        if ( printer->Open( printerName ) == TRUE )
```